File: cos10009/W11/11.4D - C Name Tester/name_tester.c
#include <stdio.h>
#include <string.h>
#include "terminal_user_input.h"

#define LOOP_COUNT 60

void print_silly_name(my_string name);

int main()
{
	my_string name;
	name = read_string("What is your name? ");

	if (strcmp(name.str, "Fred") == 0 || strcmp(name.str, "Ted") == 0) {
		printf("Your name %s is an awesome name!\n", name.str);
	} else {
		print_silly_name(name);
	}

	return 0;
}

void print_silly_name(my_string name) {
	printf("Your name %s is a", name.str);
	int i;
	for (i = 0; i < LOOP_COUNT; i++) {
		printf(" silly");
	}
	printf(" name!\n");
}

File: cos10009/W3/3.1.1T M - Name Tester (First stage)/name_tester.rb
require "./input_functions";

def print_silly_name(name)
	if (name == "Ted" || name == "Fred")
		puts(name + " is an awesome name!");
	else
		puts(name + " is a silly name");
	end
end

def main()
  	name = read_string("What is your name?");
  	print_silly_name(name);
end

main();